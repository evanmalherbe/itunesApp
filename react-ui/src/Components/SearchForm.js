import React from "react";

// Import search icon image
import icon from "../searchIcon.png";

// Import React Bootstrap components
import { Form, FormControl, Col, Row } from "react-bootstrap";
import Button from "react-bootstrap/Button";

// Import bootstrap styles
import "bootstrap/dist/css/bootstrap.min.css";

// Import custom stylesheet
import "../App.css";

// Create function to display HTTP Post form in header
function SearchForm(props) {
  return (
    <div className="searchFormDiv">
      <img src={icon} alt="search icon" className="searchIconImg" />

      <Form id="searchForm">
        <Row className="align-items-center" style={{ width: "45rem" }}>
          <Col>
            <Form.Label>Search</Form.Label>
          </Col>
          <Col md={5}>
            <FormControl
              type="text"
              placeholder="e.g. Jack Johnson"
              name="searchTerm"
              onChange={props.handleChangeSearchTerm}
            />
          </Col>
          <Col>
            <Form.Label>Media:</Form.Label>
          </Col>
          <Col md={3}>
            <Form.Select
              aria-label="Select options"
              name="mediaType"
              onChange={props.handleChangeMediaType}
            >
              <option value="all">All types</option>
              <option value="movie">Movie</option>
              <option value="podcast">Podcast</option>
              <option value="music">Music</option>
              <option value="audiobook">Audiobook</option>
              <option value="short film">Short Film</option>
              <option value="tv show">TV Show</option>
              <option value="software">Software</option>
              <option value="ebook">eBook</option>
            </Form.Select>
          </Col>
          <Col>
            <Button
              variant="primary"
              type="button"
              onClick={props.handleSubmit}
            >
              Search
            </Button>
          </Col>
        </Row>
      </Form>

      {/* End of searchFormDiv*/}
    </div>
  );
}

// Export component so it can be used by App.js
export default SearchForm;
