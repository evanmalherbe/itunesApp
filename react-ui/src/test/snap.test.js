import React from "react";
import SearchForm from "../Components/SearchForm";
import renderer from "react-test-renderer";

test("renders correctly", () => {
  const tree = renderer.create(<SearchForm />).toJSON();
  expect(tree).toMatchSnapshot();
});
