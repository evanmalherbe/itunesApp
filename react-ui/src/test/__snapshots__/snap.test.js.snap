// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="searchFormDiv"
>
  <img
    alt="search icon"
    className="searchIconImg"
    src="searchIcon.png"
  />
  <form
    className=""
    id="searchForm"
  >
    <div
      className="align-items-center row"
      style={
        Object {
          "width": "45rem",
        }
      }
    >
      <div
        className="col"
      >
        <label
          className="form-label"
        >
          Search
        </label>
      </div>
      <div
        className="col-md-5"
      >
        <input
          className="form-control"
          name="searchTerm"
          placeholder="e.g. Jack Johnson"
          type="text"
        />
      </div>
      <div
        className="col"
      >
        <label
          className="form-label"
        >
          Media:
        </label>
      </div>
      <div
        className="col-md-3"
      >
        <select
          aria-label="Select options"
          className="form-select"
          name="mediaType"
        >
          <option
            value="all"
          >
            All types
          </option>
          <option
            value="movie"
          >
            Movie
          </option>
          <option
            value="podcast"
          >
            Podcast
          </option>
          <option
            value="music"
          >
            Music
          </option>
          <option
            value="audiobook"
          >
            Audiobook
          </option>
          <option
            value="short film"
          >
            Short Film
          </option>
          <option
            value="tv show"
          >
            TV Show
          </option>
          <option
            value="software"
          >
            Software
          </option>
          <option
            value="ebook"
          >
            eBook
          </option>
        </select>
      </div>
      <div
        className="col"
      >
        <button
          className="btn btn-primary"
          disabled={false}
          type="button"
        >
          Search
        </button>
      </div>
    </div>
  </form>
</div>
`;
